tool/fakerelay/main.go

```diff
@@ -42,25 +47,29 @@ func main() {
 	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
 		start := time.Now()
 		requestCount.Add(1)
-		firstRequestOnce.Do(func() {
-			if r.Header.Get("Content-Encoding") == "gzip" {
-				reader, err := gzip.NewReader(r.Body)
-				if err != nil {
-					fmt.Fprintln(os.Stderr, err)
-					http.Error(w, err.Error(), 500)
-					return
-				}
-				r.Body = reader
-			}
-			b, err := httputil.DumpRequest(r, true)
+
+		if r.Header.Get("Content-Encoding") == "gzip" {
+			reader, err := gzip.NewReader(r.Body)
 			if err != nil {
 				fmt.Fprintln(os.Stderr, err)
 				http.Error(w, err.Error(), 500)
 				return
 			}
-			firstRequest.Set(string(b))
-			sdkInfo = ParseSDKInfo(b)
-		})
+			r.Body = reader
+		}
+
+		b, err := httputil.DumpRequest(r, true)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			http.Error(w, err.Error(), 500)
+		} else {
+			bytesRecieved.Add(int64(len(b)))
+			firstRequestOnce.Do(func() {
+				firstRequest.Set(string(b))
+				sdkInfo = ParseSDKInfo(b)
+			})
```

Comment:
Have you tried using `r.ContentLength` instead of actually reading the body of every incoming request?

Also, there's a difference between what `r.ContentLength` reports and a count of bytes of the dump. Note that `httputil.DumpRequest` is not the actual bytes including headers as it may seen. It is just a HTTP/1.1 representation used for debugging.

Reply:
switched to `r.ContentLength` 👍